app/src/main/java/org/dhis2/usescases/enrollment/EnrollmentActivity.kt

```diff
@@ -374,9 +373,7 @@ class EnrollmentActivity : ActivityGlobalAbstract(), EnrollmentView {
             if (!presenter.hasAccess()) {
                 presenter.finish(mode)
             } else {
-                formView.requestDataIntegrityCheck().observe(this) { result ->
-                    if (result is SuccessfulResult) presenter.finish(mode)
```

Comment:
Check this class.
The finish method is called here and in a callback from the formview. This was causing problems and can be the root of the test to fail.

Reply:
but this methods returns a LiveData with the result, right?. Where are you handling the result of the integrity check?